src/hierarchy_builder.rs

```diff
@@ -0,0 +1,206 @@
+extern crate geo;
+
+use zone::{Zone, ZoneIndex};
+use gst::rtree::{RTree, Rect};
+use ordered_float::OrderedFloat;
+use geo::Bbox;
+use geo::boundingbox::BoundingBox;
+
+pub struct ZonesTree {
+    pub tree: RTree<ZoneIndex>,
+}
+
+impl ZonesTree {
+    pub fn new() -> ZonesTree {
```

Comment:
really minor, but if the constructor takes no arges, it's better to implement the `Default` trait